clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -296,6 +296,8 @@ KeyStore load() throws GeneralSecurityException, IOException {
                 char[] passwordChars = password != null ? password.value().toCharArray() : null;
                 ks.load(in, passwordChars);
                 return ks;
+            } catch (GeneralSecurityException | IOException e) {
+                throw new KafkaException("Failed to load SSL keystore " + path + " of type " + type, e);
```

Comment:
Should we use try with resources? Also, do we ever want to return IOException? That's still in the signature. We should probably document the exceptions thrown as well.

Reply:
@ijuma Thanks for the review. Changed to try-with-resources, removed IOException and added javadoc.